src/dbnode/topology/testutil/topology.go

```diff
@@ -101,3 +106,41 @@ func (v TopologyView) Map() (topology.Map, error) {
 
 	return topology.NewStaticMap(opts), nil
 }
+
+// SourceAvailableHost is a human-friendly way of constructing
```

Comment:
suggestion(take/leave): i constructed topologies using something like --

```
func TestFetchTaggedResultsAccumulatorAnyResponseShouldTerminateConsistencyLevelOneSimpleTopo(t *testing.T) {
	// rf=3, 30 shards total; three identical hosts
	topoMap := tu.MustNewTopologyMap(3, map[string][]shard.Shard{
		"testhost0": tu.ShardsRange(0, 29, shard.Available),
		"testhost1": tu.ShardsRange(0, 29, shard.Available),
		"testhost2": tu.ShardsRange(0, 29, shard.Available),
	})
```

I think something equivalent `tu.MustNewStateSnapshot(...)` might read better when using than this method.

Reply:
Yeah this looks good, at this point I'd be re-writing a bunch of tests for a fairly superficial change (tests other than the ones I'm modifying in this P.R) so I'm gonna leave for now

Reply:
More than the superficial, it's about keeping the tests simpler to read/maintain. If you wanna do it in another PR, mind opening an issue

Reply:
#909